### PR TITLE
Add support for transactions

### PR DESCRIPTION
**Pull request by WhoSoup**
In a similar vein to #140, this changes the factom api to return a struct containing the data rather than just returning a JSON string. Includes a basic unit test.

### CODE REVIEW

**Comment by WhoSoup** [comment]
Decided to add support for decoding the transactions in the FBlock as well, rather than just having them be raw json.

File: transaction_pending.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+// Block.
+func GetPendingTransactions() ([]PendingTransaction, error) {
+	req := NewJSON2Request("pending-transactions", APICounter(), nil)
+	resp, err := factomdRequest(req)
+
+	if err != nil {
+		return nil, err
+	}
+	if resp.Error != nil {
+		return nil, err
```
Comment (PaulBernier) [inline transaction_pending.go:29]: return resp.Error? Because err is nil at this point

File: fblock.go [hunk 2]
```diff
@@ -8,24 +8,152 @@ import (
...
+	Outputs        []TransactionAddress `json:"outputs"`
+	OutECs         []TransactionAddress `json:"outecs"`
+	RCDs           []string             `json:"rcds"`
+	SigBlocks      []rawSigBlock        `json:"sigblocks"`
+}
+type rawSigBlock struct {
+	Signatures []string `json:"signatures"`
+}
+
+func (t *FBTransaction) MarshalJSON() ([]byte, error) {
```
Comment (PaulBernier) [inline fblock.go:76]: are MarshalJSON and UnmarshalJSON used anywhere?
And can you add unit test for those methods?

File: fblock.go [hunk 3]
```diff
@@ -8,24 +8,152 @@ import (
...
+	t.Outputs = txResp.Outputs
+	t.OutECs = txResp.OutECs
+	t.TxID = txResp.TxID
+
+	// catch decoding errors or malicious data
+	if len(txResp.Inputs) != len(txResp.RCDs) || len(txResp.Inputs) != len(txResp.SigBlocks) {
+		return fmt.Errorf("invalid signature counts")
+	}
+
+	for i := range txResp.Inputs {
```
[the remark this answers is not in the document]
Comment (WhoSoup) [reply fblock.go:113]: I needed the index anyway to access the respective `tx.RCDs` and `tx.SigBlocks`, so I decided to just use the index for all three to make the association clearer.

File: transaction_pending.go (hunk 1 above)
Comment (WhoSoup) [reply transaction_pending.go:29]: Good catch

File: fblock.go (hunk 2 above)
Comment (WhoSoup) [reply fblock.go:76]: They are used recursively by Marshalling/Unmarshalling FBlock.Transactions. UnmarshalJSON has unit test coverage due to TestUnmarshalFBlock but it appears MarshalJSON does not, which I'll add.

**Comment by WhoSoup** [comment]
Addressed the comments. Great call on the MarshalJSON unit test, managed to catch a bug where it didn't convert the timestamp properly back to millitimestamp.

**Pull request merged**